File: Hard/No_0145_Binary Tree Postorder Traversal/145_.rb
#Runtime: 36 ms, faster than 39.06% of Ruby online submissions for Binary Tree Postorder Traversal.
#Memory Usage: 9.4 MB, less than 100.00% of Ruby online submissions for Binary Tree Postorder Traversal.

# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} root
# @return {Integer[]}
def postorder_traversal(root)
    curr = []
    return helper(root, curr)
end


def helper(root, curr)
    if root
        helper(root.left, curr)
        helper(root.right, curr)
        curr.push root.val
    end
    return curr
end
    

File: Hard/No_0145_Binary Tree Postorder Traversal/145_.js

//Runtime: 52 ms, faster than 79.50% of JavaScript online submissions for Binary Tree Postorder Traversal.
//Memory Usage: 33.8 MB, less than 50.00% of JavaScript online submissions for Binary Tree Postorder Traversal.


/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var postorderTraversal = function(root) {
    let curr = [];
    return helper(root, curr)
    
};

var helper = function(root, curr){
    if(root){
        helper(root.left, curr);
        helper(root.right, curr);
        curr.push(root.val);
    }
    return curr
}
